tests/e2e/vmgroup_volumeops_test.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test is going to create volume on the fresh testbed very first time.
+// After installing vmdk volume plugin/driver, volume creation should not be
+// failed very first time.
+
+// This test is going to cover the issue reported at #656
+// TODO: as of now we are running the test against photon vm it should be run
+// against various/applicable linux distros.
+
+package e2e
+
+import (
+	"os"
+	"testing"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+)
+
+const (
+	volName1                 = "vmgroup_test_vol_1"
+	volName2                 = "vmgroup_test_vol_2"
+	defaultVg                = "_DEFAULT"
+	defCname                 = "def_test_ctr"
+	defVgName                = "def_test_vg"
+)
+
+// Tests to validate behavior with the __DEFAULT_ vmgroup.
+
+var vmHosts = []string{os.Getenv("VM1"), os.Getenv("VM2")}
+var host, vmgroup, ds1, ds2, cname string
+
+func getEnv(t *testing.T) {
+	host = os.Getenv("ESX")
+	vmgroup = os.Getenv("T1")
+	ds1 = os.Getenv("DS1")
+	ds2 = os.Getenv("DS2")
+	cname = os.Getenv("CNAME")
+
+	if vmgroup == "" {
+		vmgroup = defVgName
+	}
+
+	if ds1 == "" || ds2 == "" {
+		t.Errorf("Unknown datastores, please set DS1 and DS2 to valid datastore names in the environment")
+	}
+
+	if cname == "" {
+		cname = defCname
+	}
+}
+
+func createVolumeOnDefaultVg(t *testing.T, name string) {
+	// 1. Create the volume on each host
+	for _, vm := range vmHosts {
+		_, err := dockercli.CreateVolume(vm, name, "")
+		if err != nil {
+			t.Errorf("Error while creating volume - %s on VM - %s, err - %s", name, vm, err.Error())
+		}
+	}
+
+	// 2. Verify the volume is created on the default vm group
+	cmd := admincli.ListVolumes + " | grep " + defaultVg + " | grep " + name
+	val, err := ssh.ExecCmd(host, cmd)
+	if err != nil {
+		t.Errorf("Error while listing volumes [%s] in default vmgroup on host %s [%s] err - %s", cmd, host, err.Error())
+	}
+	if val == "" {
+		t.Errorf("Volume %s not found in default vmgroup", name)
+	}
+}
+
+// TestVolumeCreateOnDefaultVg - Verify that volumes can be created on the
+// default volume group with default permissions, then attached and deleted
+func TestVolumeCreateAndAttachOnDefaultVg(t *testing.T) {
```

Comment:
To keep it consistent (reference of any existing test), high level steps should be added per test method.